### PR TITLE
Return HTTP OK on token drop for oauth2 authentication

### PR DESCRIPTION
**Pull request by wendigo**
HttpTokenPoller waits 4 seconds for token drop to be acknowledged (checks for HTTP OK). With this small change CLI/JDBC returns token almost immediately upon resolution.

### CODE REVIEW

**Review by kokosing — APPROVED** [review]
Why it was waiting?

**Comment by wendigo** [comment]
@kokosing because HttpTokenPoller waits for the ACK (HTTP OK) on HTTP DELETE on token endpoint with a timeout of 4s. And since we didn't return Response http status was not HTTP OK (I guess that it was NO CONTENT)

**Comment by kokosing** [comment]
> HttpTokenPoller waits for the ACK (HTTP OK)

Can we make HttpTokenPoller to wait for 2xx?

Are there any other places likes that?

**Comment by wendigo** [comment]
@kokosing done

File: client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java [hunk 1]
```diff
@@ -96,7 +97,7 @@ public void tokenReceived(URI tokenUri)
                     .withMaxAttempts(-1)
                     .withMaxDuration(Duration.ofSeconds(4))
                     .withBackoff(100, 500, MILLIS)
-                    .handleResultIf(code -> code != HTTP_OK))
+                    .handleResultIf(code -> code >= HTTP_INTERNAL_ERROR))
```
Comment (kokosing) [inline client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java:100]: what about 3xx and 4xx?

Comment (s2lomon) [reply client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java:100]: it should always be a kind of fire and forget, so there is no point in retrying for 300 or 400 as we won't handle that either way.

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java [hunk 2]
```diff
@@ -114,13 +114,16 @@ private static Response pendingResponse(HttpServletRequest request)
...
-    public void deleteAuthenticationToken(@PathParam("authId") UUID authId)
+    public Response deleteAuthenticationToken(@PathParam("authId") UUID authId)
     {
         if (authId == null) {
             throw new BadRequestException();
         }
 
         tokenExchange.dropToken(authId);
+        return Response
+                .ok()
```
Comment (s2lomon) [inline core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java:125]: Maybe return some `ok("Authentication dropped", Text)` as for no content 204 should be returned as it is right now.

**Review by s2lomon — APPROVED** [review]
% 1 minor comment.

File: client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java (hunk 1 above)
Comment (wendigo) [reply client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java:100]: Exactly. There is no point in retrying 300 and 400s

File: core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java (hunk 2 above)
Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchangeResource.java:125]: I don't think that's necessary. Human does not consume this endpoint and we check only for http status

**Comment by wendigo** [comment]
Ping @kokosing

**Pull request merged**